clang/lib/CodeGen/CodeGenPGO.cpp

```diff
@@ -239,9 +239,12 @@ struct MapRegionCounters : public RecursiveASTVisitor<MapRegionCounters> {
     if (MCDCMaxCond == 0)
       return true;
 
-    /// At the top of the logical operator nest, reset the number of conditions.
-    if (LogOpStack.empty())
+    /// At the top of the logical operator nest, reset the number of conditions,
+    /// also forget previously seen split nesting cases.
+    if (LogOpStack.empty()) {
       NumCond = 0;
+      SplitNestedLogicalOp = false;
```

Comment:
Reset of SplitNestedLogicalOp could also probably be done in dataTraverseStmtPost() at the point where the diagnostic is emitted, but this is OK.

Reply:
Definitely a bug that we weren't resetting this.